### PR TITLE
Transaction empty description -> "no description"

### PR DESCRIPTION
**Pull request by BerghsJelmer**
Not sure if this works, or if this is the place to put it.

Fixes issue #6210

Changes in this pull request:

- Checks for empty description, if it is empty, fill in default: "(no description)"

@JC5

### CODE REVIEW

File: app/Services/Shared/Import/Routine/ApiSubmitter.php [hunk 1]
```diff
@@ -359,6 +359,9 @@ private function replaceMappings(array $line): array
                         app('log')->debug(sprintf('Replaced source name "%s" with a reference to account id #%d', $source, $this->mapping[0][$source]));
                     }
                 }
+                if (empty(trim($transaction['description']))){
```
Comment (JC5) [inline app/Services/Shared/Import/Routine/ApiSubmitter.php:362]: I would use:

```php
if('' === trim((string)($transaction['description'] ?? '')) {
```

`empty()` will return true when the value is `0` or `"0"`, but without `empty()` you need to catch the (tiny) chance that `['description']` isn't set.

Sounds good?

**Comment by BerghsJelmer** [comment]
pushed a new commit, tested and it works now.

**Comment by JC5** [comment]
So why not move the code to `getDescription()`?

**Comment by JC5** [comment]
Or leave it where it was as a safety catch. Either way.

**Comment by BerghsJelmer** [comment]
What do you prefer?
I can move it

**Comment by BerghsJelmer** [comment]
Could this pr get merged?

**Comment by JC5** [comment]
I was waiting for the original code to be restored.

**Comment by JC5** [comment]
Since I'm working on the code right now, I merged the commit so you get the credits, and have taken your original code to the right place. I'm sure I could do this in your branch but I have no idea how.

Either way, thanks for your support and code contribution; it's much appreciated!

**Pull request merged**